src/inference/src/cache_manager.hpp

```diff
@@ -99,16 +99,32 @@ class FileStorageCacheManager final : public ICacheManager {
 
 private:
     void write_cache_entry(const std::string& id, StreamWriter writer) override {
+        // Fix the bug caused by pugixml, which may return unexpected results if the locale is different from "C".
+        auto original_plocal = setlocale(LC_NUMERIC, nullptr);
+        if (std::strcmp(original_plocal, "C") != 0) {
```

Comment:
I suppose we can change it unconditionally ? anyway, we restore it after

Reply:
BTW, do we need the same for IR reader / serializer? It also operates with numbers

Reply:
the reader / serializer is called by write_cache_entry() & read_cache_entry(), right?
If yes, this setlocale(LC_ALL, "C"); will take effect for them.